mlrun/artifacts/document.py

```diff
@@ -130,15 +130,16 @@ def lazy_load(self) -> Iterator["Document"]:  # noqa: F821
                 artifact = self.producer.log_document(
                     key=self.artifact_key,
                     document_loader=self.loader_spec,
-                    src_path=self.source_path,
+                    local_path=self.local_path,
                     upload=self.upload,
                 )
-                yield artifact.to_langchain_documents()
+                res = artifact.to_langchain_documents()
+                yield res[0]
 
         # Return an instance of the dynamically defined subclass
         instance = DynamicDocumentLoader(
             artifact_key=artifact_key,
-            source_path=source_path,
+            local_path=source_path,
             loader_spec=loader_spec,
             producer=producer,
             upload=upload,
```

Comment:
Repeating my comment from previous PR (regarding the `artifact_key_instance` method), let's discuss - 
Hex values are a bit distracting here, resulting in things like `22tmp22myfile44txt` where it's hard to separate the file-name from the filled chars. Why not simply replace with -? Will result in something like -tmp-myfile-txt in this case.

Reply:
Then we loose uniqueness - /tmp/file and /tmp~file will generate same key

Reply:
Indeed, but that's a pretty much of a corner case, don't you think? I think it's better to have better readability for the common case than sacrifice it for some cases that may never happen.

Reply:
Hmm, I disagree - when such a thing happens (and it will), you won't be able to find the problem. As a result, you will have an artifact with wrong content, and worse yet, you won't even know about it.

Reply:
I have to say that I feel that users managing their files this way are bound to get confused one way or the other. BTW, this will cause issues only if they choose to upload the files to the MLRun store (otherwise the content is not handled by MLRun).